CONTRIBUTING.md

```diff
@@ -0,0 +1,201 @@
+
+1. Go - Rich Go language support for Visual Studio Code;
+2. vscode-proto3 - Protobuf 3 support for Visual Studio Code;
+3. GitLens - Git supercharged;
+
+Update the Go extension settings: `File -> Preferences -> Settings : @ext:golang.go`
+
+- Set `Go: Lint Tool` to `golangci-lint`;
+- Set `Go: Format Tool` to `gofumpt`;
+
+_The required Go Tools may need to be downloaded by your IDE._
+
+For __CosmWasm__ and _the language that shall not be named_, please refer to the [Archway Developer Portal](https://docs.archway.io/developers).
+
+## 2. Issues
+
+Public collaboration and coordination are driven via issues. For a brief overview of using GitHub Issues, [please read this](https://docs.github.com/en/issues/tracking-your-work-with-issues/quickstart).
+
+1. [Find](https://github.com/archway-network/archway/issues) or [Open](https://github.com/archway-network/archway/issues/new) an issue;
+2. __Engage__ in _constructive_ and _thoughful_ __discussion__ on the issue;
+3. __Coordinate__ with _other contributors_ on the issue;
+4. __Follow__ _GitHub best practices_ and the _guidelines_ in the issue templates;
+
+This repository uses the __good first issue__ label to identify issues suitable for first-time contributors.
+
+Ensure a proper title for the issue that clearly defines what the issue is. This will greatly assist others to easily find related issues and avoid duplication.
+
+Make use of the following labels to provide additional context:
+
+- _bug_: Something isn't working;
+- _documentation_: Improvements or additions to documentation;
+- _enhancement_: New feature or request;
+- _question_: Further information is requested;
+
+Example: `Lack of Community Guidelines` <- label as `documentation`
+
+## 3. Branching
+
+The `main` branch represents the latest development features and fixes. Thus, it will most likely not be compatible with running networks such as `mainnet` or any testnets.
+
+Releases are found under `release/#.#.x` using a semantic versioning scheme {braking}.{feature}.{fix} denoting each minor version.
+
+The `main` branch and `release/*` branches are protected and can only be updated via a Pull Request. Thus, do not make any commits to these branches directly.
+
+For increased manageability please address one issue per branch and use the following branch name format:
+
+`<user>/<issue#>-<short_description>`
+
+## 4. Testing
+
+All _features_ and _fixes_ must be accompanied by appropriate unit/integration tests. Please ensure to follow Go and CosmosSDK best practices regarding tests. Archway utilizes table-driven tests, as explained in [this blogpost](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) by Dave Cheney.
+
+Simple rules for table-driven tests:
+
+1. Keep tests independent (ordering of tests should not impact testing);
+2. Test one construct or aspect per test;
+3. Every test should make it obvious what is being tested;
+
+## 5. Pull Requests
+
+* All pull requests must target `main`;
```

Comment:
Not sure if this would apply to all scenarios.  Some features which are big enough are usually developed in dedicated feature branches before being merged to main.